pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -320,15 +323,18 @@ func getNICIPAddresses(ctx context.Context, c *config, ifaceName string) ([]stri
 				if err != nil {
 					return nil, fmt.Errorf("failed to retrieve IP string for IP %q: %v", name, err)
 				}
-				ipAddresses = append(ipAddresses, publicIPs...)
+				for _, ip := range publicIPs {
+					ipAddresses[ip] = v1.NodeExternalIP
+				}
 			}
 
 			internalIPs, err := getInternalIPAddresses(ctx, c, ifaceName, name)
 			if err != nil {
 				return nil, fmt.Errorf("failed to retrieve internal IP string for IP %q: %v", name, err)
 			}
-
-			ipAddresses = append(ipAddresses, internalIPs...)
+			for _, ip := range internalIPs {
+				ipAddresses[ip] = v1.NodeExternalIP
```

Comment:
Should we use `v1.NodeInternalIP` since we are talking about internal IPs? Or it works a bit differently for Azure?

Reply:
You are right, copy/paste mistake :)